### PR TITLE
How Privacy Computation Engine Works

### PR DESCRIPTION
**Pull request by milstan**
High-level explainer on how Privacy Computation Egine and Privacy Compiler work.

Covers configuration and what to expect from those components. Also specifies all the fields necessary to configure the PCE (including RoPA)

Close blindnet-io/devrel-management/issues/52
Close blindnet-io/devrel-management/issues/51
Close blindnet-io/product-management/issues/829

### CODE REVIEW

**Comment by milstan** [comment]
I wrote basic black-box-view documentaiton on how Privacy Computation Engine and Privacy Compiler work.

I used the "How to configure..." part do define the settings kind of input the PCE takes. @m4rk055 please review this ASAP and see if it is in sync with what you are doing. Feel free to make changes if not, but let us itterate this week to have a stable definition of what are the settings that go in.

My idea is to have the Systems define a table of records that can be easily derived from typical privacy policies. See for example [this one](https://en.feed.co/pages/privacy-policy) "7 - Treatment details". The whole setting should be easily defined by reading the privacy policy. Also, it should be as easy, given the settings, to generate a Ptivacy policy.

p.s. Also I don't let the System specify their own human-readable lables for PRIV categories and other terms here. I assume, Systems can have a configuration file where they handle localized labels for PRIV terms, and then they can here, in configuration, only provide the PRIV Term.

File: docs/how_it_works/devkit/pce.mdx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+## Configuring your Privacy Computation Engine
+
+You must first configure the Privacy Computation Engine in order to make the most of it.
+It needs to know about:
+- General information about your Organization and its' data processing policies (Equivalent to a RoPA - Record of Processing Activities)
+- Mappings of your data structures to [PRIV](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/RFC-PRIV.md) Data Categories
+- Few choices you have to make about when to authenticate the user (before or after they make a Privacy Request), and when to deliver automatic responses and when to wait for validation
+
+To Configure your Privacy Computation Engine prepare the following information:
```
Comment (m4rk055) [inline docs/how_it_works/devkit/pce.mdx:11]: Also `data consumer categories` according to [ROPA](https://transcend.io/record-of-processing-activities-ropa/)

File: docs/how_it_works/devkit/pce.mdx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+| If you are processing data on behalf of someone else (a Controller), then names, contact details of all such Controllers their representatives and the data protection officers | `string` | see above |
+| A link to your Privacy Policy | `URL` | `https://blindnet.io/privacy` |
+| A general description of the technical and organisational security measures referred to in Article 32 of GDPR | `string` | `We use administrative, technical, and physical safeguards to protect your personal data, taking into account the nature of the personal data and the processing, and the threats posed.`|
+| Countries where data servers are located (included those of the processors who are processing data on your behalf) | `string` | `France` |
+| A list of Retention Policies | [Retention Policies](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/RFC-PRIV.md#retention-policy) | `data-categories`:`CONTACT`; `policy-type`:`NO-LONGER-THAN`, `duration`: `180 days`, `after`: `RELATIONSHIP-END` |
```
Comment (m4rk055) [inline docs/how_it_works/devkit/pce.mdx:20]: I'd start by defining selectors (data category for each).
For each data category/selector, define list of provenances and list of retention policies.

Then define legal bases (type, name, description, privacy scope).

Having those 2, **Processing Records** can be inferred.

**Review by m4rk055 — CHANGES_REQUESTED** [review]

File: docs/how_it_works/devkit/pce.mdx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+It needs to know about:
+- General information about your Organization and its' data processing policies (Equivalent to a RoPA - Record of Processing Activities)
+- Mappings of your data structures to [PRIV](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/RFC-PRIV.md) Data Categories
+- Few choices you have to make about when to authenticate the user (before or after they make a Privacy Request), and when to deliver automatic responses and when to wait for validation
+
+To Configure your Privacy Computation Engine prepare the following information:
+| Information | Format | Example|
+| :----: | :-----: | :--------|
+| Name and contact details of your Organization and its representative | `string` | `Blindnet Inc. hello@blindnet.io Vuk Janosevic`|
+| Identity and contact of a Data Protection Officer - if you are using blindnet devkit Privacy Request Builder, include the URL where you are hosting the interface | `string` | `Vuk Janosevic, www.blindnet.io/privacy-request-builder`|
```
[the remark this answers is not in the document]
Comment (milstan) [reply docs/how_it_works/devkit/pce.mdx:15]: same as Org

File: docs/how_it_works/devkit/pce.mdx (hunk 2 above)
Comment (milstan) [reply docs/how_it_works/devkit/pce.mdx:20]: I am OK with that as well. Feel free to change accordingly.

The question is really what is easier for the peson configuring the system to define? Is it easier to reason in terms of provenances, lgeal bases, retention policies (and have RoPA derived) or int erms of RoPA (and have the provenances, lgeal bases, retention policies derived).

I think indeed RoPA is a bit of an articial cosntruct.

File: docs/how_it_works/devkit/pce.mdx (hunk 1 above)
Comment (milstan) [reply docs/how_it_works/devkit/pce.mdx:11]: True. Added.

File: docs/how_it_works/devkit/pce.mdx (hunk 2 above)
Comment (milstan) [reply docs/how_it_works/devkit/pce.mdx:22]: I made the change as you suggested. 

A mapping of database-field : selector is provided.

[Provenances](https://github.com/blindnet-io/product-management/blob/priv-patch/refs/schemas/priv/RFC-PRIV.md#provenance) and [Retention Policies](https://github.com/blindnet-io/product-management/blob/priv-patch/refs/schemas/priv/RFC-PRIV.md#retention-policy) already include Data Category/Selector in the new PRIV patch, so it is enought just to give a list of those objects.

**Review by m4rk055 — APPROVED** [review]

**Comment by noelmace** [comment]
The PR has been approved by @m4rk055, but https://github.com/blindnet-io/blindnet.dev/pull/27/commits/c1554a54c9e9e21b15714c5a458197e85add1c8c introduced several changes afterwards. I'm therefore asking for a new approval before merging it.

**Review by m4rk055 — APPROVED** [review]

**Pull request merged**